lib/endpoints/class-wp-rest-coauthors-authorposts-endpoint.php

```diff
@@ -83,41 +72,62 @@ public function get_items_permissions_check( $request ) {
 				return new WP_Error( 'rest_post_invalid_id', __( 'Invalid post id.' ), array( 'status' => 404 ) );
 			}
 
-			if ( ! $this->parent_controller->check_read_permission( $parent ) ) {
-				return new WP_Error( 'rest_forbidden', __( 'Sorry, you cannot view this post.' ), array( 'status' => rest_authorization_required_code() ) );
-			}
-
-			$post_type = get_post_type_object( $parent->post_type );
-			if ( ! current_user_can( $post_type->cap->edit_post, $parent->ID ) ) {
-				return new WP_Error( 'rest_forbidden', __( 'Sorry, you cannot view the authors for this post.' ), array( 'status' => rest_authorization_required_code() ) );
-			}
-
 			return true;
 		}
 
 		return true;
 	}
 
 	/**
-	 * Check if a given request has access to create a authors entry for a post.
+	 * Check if a given request has access to create an author association for a post.
 	 *
 	 * @param WP_REST_Request $request Full data about the request.
 	 *
 	 * @return boolean
 	 */
 	public function create_item_permissions_check( $request ) {
-		return $this->get_items_permissions_check( $request );
+		if ( ! empty( $request['parent_id'] ) ) {
+			$parent = get_post( (int) $request['parent_id'] );
+
+			if ( empty( $parent ) || empty( $parent->ID ) ) {
+				return new WP_Error( 'rest_post_invalid_id', __( 'Invalid post id.' ), array( 'status' => 404 ) );
+			}
+
+			$post_type = get_post_type_object( $parent->post_type );
+			if ( ! current_user_can( $post_type->cap->edit_post, $parent->ID ) ) {
+				return new WP_Error( 'rest_forbidden', __( 'Sorry, you cannot create an author association for this post.' ), array( 'status' => rest_authorization_required_code() ) );
+			}
+
+			return true;
+		}
+
+		return new WP_Error( 'rest_forbidden', __( 'Creating an author is not supported.' ), array( 'status' => rest_authorization_required_code() ) );
 	}
 
 	/**
-	 * Check if a given request has access to update a authors entry for a post.
+	 * Check if a given request has access to update the author association for a post.
 	 *
 	 * @param WP_REST_Request $request Full data about the request.
 	 *
 	 * @return boolean
 	 */
 	public function update_item_permissions_check( $request ) {
-		return $this->get_items_permissions_check( $request );
+		if ( ! empty( $request['parent_id'] ) ) {
+			$parent = get_post( (int) $request['parent_id'] );
+
+			if ( empty( $parent ) || empty( $parent->ID ) ) {
+				return new WP_Error( 'rest_post_invalid_id', __( 'Invalid post id.' ), array( 'status' => 404 ) );
+			}
+
+			$post_type = get_post_type_object( $parent->post_type );
+			if ( ! current_user_can( $post_type->cap->edit_post, $parent->ID ) ) {
```

Comment:
Shouldn't be passing `$parent->ID` here. `current_user_can` only has 1 param: https://core.trac.wordpress.org/browser/tags/4.6/src/wp-includes/capabilities.php#L447